nomos-services/data-availability/network/src/backends/libp2p/executor.rs

```diff
@@ -147,7 +145,13 @@ where
         let dispersal_blobs_sender = executor_swarm.dispersal_blobs_channel();
         let executor_open_stream_sender = executor_swarm.dispersal_open_stream_sender();
 
-        let task = overwatch_handle.runtime().spawn(executor_swarm.run());
+        let (task_abort_handle, abort_registration) = AbortHandle::new_pair();
+        let task = (
+            task_abort_handle,
+            overwatch_handle
+                .runtime()
+                .spawn(Abortable::new(executor_swarm.run(), abort_registration)),
```

Comment:
Just curious, if in future we are planning to do some draining or wait for underlying tasks to complete, how could we inform the inner executor or validator swarms that it's been canceled and should wrap up it's work?